src/ShopBundle/objects/TShopBasket/TShopBasketVoucherCoreList.class.php

```diff
@@ -11,6 +11,9 @@
 
 class TShopBasketVoucherCoreList extends TIterator
 {
+
+    protected $checkedAndValid = [];
```

Comment:
darf das auch private sein?

Reply:
Yes it can. I had it protected because I initially thought that children might need to track this state, but this wasn't the case in the end.